### DISCUSSION TITLE
Unable to Duplicate Meep Python Mie Scattering Tutorial

### DISCUSSION THREAD

**Discussion by cgbsu** [NanoComp/meep#2590] [General]
Hello, 

    I am having some trouble duplicating the [Mie Scattering section](https://meep.readthedocs.io/en/latest/Python_Tutorials/Basics/#mie-scattering-of-a-lossless-dielectric-sphere) of the tutorial. My analytical and numerical results are definitely not in agreement. I dont know a much about Mie Scattering, and I am finding it hard to figure out how to diagnose the problem from a physics perspective.

I wrote my version of the tutorial differently, but I have been over the code a number of times, but it still does not come out right.

I am linking the [notebook](https://github.com/cgbsu/share-notebooks/blob/main/meep_tutorial_0_part_6.ipynb), you can see clearly the results are way off and see my code.

I tried a couple of suggestions I saw on different Issues/threads from this repo, like disabling the symmetry, but it did not work.

The main difference with this code is the introduction of fluxes/flux regions is procedural, and simulators are automatically generated. All fluxes and flux data arrays are logged. Also fluxes are automatically subtracted.

It seems no combination of fluxes from the control (no geometry) and variable (sphere geometry) or the fluxes themselves changes the qualitative results.

There are more results in the [notebook](https://github.com/cgbsu/share-notebooks/blob/main/meep_tutorial_0_part_6.ipynb) I linked, but this is the result I think should be correct: 
![image](https://github.com/NanoComp/meep/assets/47073324/bd3d04a5-8682-4e5c-ae7f-671e1b175cf1)

The different "Meep" lines are essentially as if I had changed 
```python
box_x1_flux0 = mp.get_fluxes(box_x1)
```
to be a different flux box in the original

Any suggestions?

Thanks!

**Comment by smartalecH** [comment]
As a sanity check, when you run the existing tutorial as is, you can replicate the reported results, right?

**Reply by cgbsu** [reply]
Thanks for the reply, its a tad off actually 
![Figure 1(13)](https://github.com/NanoComp/meep/assets/47073324/ac243c91-ac6a-47c6-b51e-e9711152f6ba)

**Reply by cgbsu** [reply]
I also discovered I was running the control with `until_after_sources` set to `100` for both sims, but now that I am doing `10` and `100` it looks somewhat different, maybe a bit closer
![Figure 10(2)](https://github.com/NanoComp/meep/assets/47073324/da667a8a-4388-4354-8e6e-2e093ace007d)
![Figure 12](https://github.com/NanoComp/meep/assets/47073324/28cb09e8-22b3-4a07-91ce-b3cb74ba6afb)
![Figure 8(1)](https://github.com/NanoComp/meep/assets/47073324/52718134-369f-47bf-8d63-bad2b6470a4d)

**Reply by cgbsu** [reply]
Hello, just wanted to follow up

**Comment by dalarev** [comment]
See if [this](https://github.com/NanoComp/meep/discussions/1911) similar discussion helps answer your questions. 
At the very least, it could help explain why a copy/paste of the tutorial code doesn't produce the exact same results.
